flink-kubernetes/src/test/java/org/apache/flink/kubernetes/KubernetesResourceManagerTest.java

```diff
@@ -323,6 +330,18 @@ public void testGetCpuCoresNumSlots() {
 		assertThat(resourceManager.getCpuCores(configuration), is(3.0));
 	}
 
+	@Test
+	public void testCreateTaskManagerPodFailedAndRetry() throws Exception {
+		flinkKubeClient = spy(createFlinkKubeClient());
```

Comment:
According to Flink code style, we avoid using mockito mocks.
We usually create proper testing implementations of classes.
E.g. in this case, we could have:
```
private static FlinkKubeClient createTestingFlinkKubeClientAllocatingPodsAfter(
    int numberOfRetries,
    AtomicInteger retries) { ... }
```
which fails and increments `retries` until it reaches `numberOfRetries` and succeeds.

Reply:
Actually, i just want to reuse the `verify` with timeout in `mockito`. So i use the `spy`. But i think your suggestion makes sense, i will use the testing implementations instead.